lib/jxl/decode.cc

```diff
@@ -2549,7 +2576,23 @@ JxlDecoderStatus JxlDecoderGetFrameHeader(const JxlDecoder* dec,
   }
   header->name_length = dec->frame_header->name.size();
   header->is_last = dec->frame_header->is_last;
-
+  if (!dec->coalescing && dec->frame_header->custom_size_or_origin) {
+    header->crop_x0 = dec->frame_header->frame_origin.x0;
```

Comment:
Right!

These combinations of having layers and nontrivial orientation are not likely to occur in the wild, since at the moment jxl is the only image format that supports both layers and orientation :) - but of course we do have to make it do the right thing. How orientation affects crop positions makes me feel dizzy btw.

Reply:
OK, fixed the crop offsets in case of orientation, so in the case of !keep_orientation and non-coalescing decoding, the layers are returned with sizes and crop offsets normalized so the application can just assume that things are always identity-oriented. I added a test for this (OrientedCroppedFrameTest) to verify that in all 8 orientations, manually blending non-coalesced frames (here kept easy by using only kReplace) based on the returned offset and dimensions results in the same image as letting the default coalescing decoder do it.